storage/storage.cpp

```diff
@@ -614,14 +614,28 @@ void Storage::DownloadNextFile(QueuedCountry const & country)
 {
   TCountryId const & countryId = country.GetCountryId();
   CountryFile const & countryFile = GetCountryFile(countryId);
+  auto const opt = country.GetCurrentFileOptions();
 
-  string const filePath = GetFileDownloadPath(countryId, country.GetCurrentFileOptions());
+  string const readyFilePath = GetFileDownloadPath(countryId, opt);
   uint64_t size;
+  auto & p = GetPlatform();
+
+  // Since a downloaded valid diff file may be either with .diff or .diff.ready extension,
+  // we have to check these both cases in order to find
+  // the diff file which is ready to apply.
+  // If there is a such file we have to cause the success download scenario.
+  bool isDownloadedDiff = false;
+  if (opt == MapOptions::Diff)
+  {
+    string filePath = readyFilePath;
+    my::GetNameWithoutExt(filePath);
+    isDownloadedDiff = p.GetFileSizeByFullPath(filePath, size);
```

Comment:
Implicit convert from integral to boolean? What about empty files?

Reply:
I didn't get where you saw that. GetFileSizeByFullPath returns bool.

Reply:
Ah, see now, sorry.